File: 2022-2023/sourcecode/lab/lab5/CarsyAppJava/app/src/main/java/pl/udu/uwr/pum/carsyappjava/data/Cost.java
package pl.udu.uwr.pum.carsyappjava.data;

import java.time.LocalDate;


public class Cost {
    private final CostType type;
    private final LocalDate date;
    private final int amount;

    public Cost(CostType type, LocalDate date, int amount){
        this.type = type;
        this.date = date;
        this.amount = amount;
    }

    public CostType getType() {
        return type;
    }

    public LocalDate getDate() {
        return date;
    }

    public int getAmount() {
        return amount;
    }
}

File: 2022-2023/sourcecode/lab/lab5/CarsyAppKotlin/app/src/main/java/pl/udu/uwr/pum/carsyappkotlin/data/Cost.kt
package pl.udu.uwr.pum.carsyappkotlin.data

import pl.udu.uwr.pum.carsyappkotlin.R
import java.time.LocalDate

enum class CostType(val costType: String, val icon: Int) {
    REFUELING("Tankowanie", R.drawable.ic_fuel),
    SERVICE("Serwis", R.drawable.ic_car_repair),
    PARKING("Parking", R.drawable.ic_parking),
    INSURANCE("Ubezpieczenie", R.drawable.ic_general_cost),
    TICKET("Mandat", R.drawable.ic_ticket)
}

data class Cost (
    val type: CostType,
    val date: LocalDate,
    val amount: Int
    )